### PR TITLE
Remove duplicate fused type definition

### PR DESCRIPTION
**Pull request by rfezzani**
## Description

`integral_floating` fused type defined in `transform.pxd` and `_haar.pxd` is duplicate of `np_real_numeric` that is defined in `fused_numerics.pxd`.

This PR proposes to remove `integral_floating`.

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/feature/_cascade.pyx [hunk 1]
```diff
@@ -87,11 +87,11 @@ cdef struct Stage:
 
     Py_ssize_t first_idx
     Py_ssize_t amount
-    float threshold
+    cnp.float32_t threshold
 
 
 cdef vector[Detection] _group_detections(vector[Detection] detections,
-                                         float intersection_score_threshold=0.5,
+                                         cnp.float32_t intersection_score_threshold=0.5,
```
Comment (hmaarrfk) [inline skimage/feature/_cascade.pyx:94]: this seems like a behavior change. float used to be float64. wy the change to 32 bit? Is the change to 32 bit correct?

Comment (rfezzani) [reply skimage/feature/_cascade.pyx:94]: In my mind `float` is `float32` and `double` is `float64` in cython, but I am may be mistaken :confused:

Comment (rfezzani) [reply skimage/feature/_cascade.pyx:94]: Cython [documentation](https://cython.readthedocs.io/en/latest/src/userguide/language_basics.html#types) confirms that ;)

Comment (hmaarrfk) [reply skimage/feature/_cascade.pyx:94]: ok my confusion stems from numpy then:

```
In [1]: np.dtype(float)                                                         
Out[1]: dtype('float64')
```
I actually remember having this same discussion with jni and stefanv

**Comment by hmaarrfk** [comment]
Maybe we should upstream these definitions to numpy.

**Comment by alexdesiqueira** [comment]
Thank you @rfezzani!

**Comment by rfezzani** [comment]
Thank you @hmaarrfk and @alexdesiqueira ;)

**Pull request merged**